apps/api/src/profiles/profiles.controller.ts

```diff
@@ -1,35 +1,91 @@
-import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
+import {
+  Controller,
+  Post,
+  Body,
+  UseGuards,
+  Get,
+  Req,
+  Res,
+  HttpException,
+  HttpStatus,
+  Patch,
+} from '@nestjs/common';
+import { AuthGuard } from 'src/auth/auth.guard';
+import Session from 'supertokens-node/recipe/session';
+import Passwordless from 'supertokens-node/recipe/passwordless';
 import { ProfilesService } from './profiles.service';
 import { CreateProfileDto } from './dto/create-profile.dto';
 import { UpdateProfileDto } from './dto/update-profile.dto';
-import { Profile } from './entities/profile.entity';
 
+@UseGuards(AuthGuard)
 @Controller('users/profile')
 export class ProfilesController {
   constructor(private readonly profilesService: ProfilesService) {}
 
   @Post()
-  create(@Body() createProfileDto: CreateProfileDto): Profile {
-    return this.profilesService.create(createProfileDto);
+  async create(
+    @Req() req: any,
+    @Res({ passthrough: true }) res: any,
+    @Body() createProfileDto: CreateProfileDto
+  ) {
+    let authid: string;
+    let mobile: string;
+    try {
+      const session = await Session.getSession(req, res);
```

Comment:
Change this to session dependency injection.

https://supertokens.com/docs/passwordless/common-customizations/verify-session